File: cpp/62_Unique Paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
  
     vector<vector<int>>v(m,vector<int>(n,1)); 
     int r=findmin(0,0,m,n,v);   
     return r;
     
      
    }
    
    int findmin(int i,int j,int m,int n,vector<vector<int>>&dp){
        
        
      
        for(int i=1;i<m;i++){
            for(int j=1;j<n;j++){
        
               dp[i][j]=dp[i-1][j]+dp[i][j-1];
            }
        }
        
        return dp[m-1][n-1];
        
        
    }
};

File: py/62. Unique Paths.py
class Solution:
    def __init__(self):
        self.mem={}
    def uniquePaths(self, m: int, n: int) -> int:
        key=(m,n)
        if m==0 or n==0: return 0
        if m==1 and n==1: return 1
        if key in self.mem: return self.mem[key]
        self.mem[key]=self.uniquePaths(m-1,n)+self.uniquePaths(m,n-1)
        return self.mem[key]
